[WIP] Changelog - August (#2578)
gitpod-io/website: src/lib/contents/changelog/2022-09-01.md

---

> title: August Gitpod Release

Reviewer: I think we shoud put an year, there was some concern about duplicate titles cc @gtsiolis

Author: Added in b5a2fc34f3a4b1fa165375f160d92be3f5b84077

</script>

TODO: Insert the featured changelog entry here.

<p><Contributors usernames="" /></p>

## All VS Code Desktop users on SaaS are now using SSH Gateway! 🎉

We have now completed a migration of all VS Code Desktop users on SaaS to our new SSH Gateway approach. This change brings improved stability, smoother restarting of workspaces, and more.

To get setup, we recommend you [upload an SSH key to Gitpod](https://www.gitpod.io/docs/configure/ssh#upload-an-ssh-key-to-gitpod). For full details on why we made this change to our SSH approach for VS Code Desktop, please check out our blog post [VS Code Desktop and SSH explained](https://www.gitpod.io/blog/vscode-desktop-ssh-updates).

If you cannot use the new approach for any reason, please reach out via the GitHub issue #[7452](https://github.com/gitpod-io/gitpod/issues/7452).

## View Resource Consumption in JetBrains Backend Control Center

When using a Gitpod workspace you might experience performance issues caused by an application using more resources than expected, or due to resource consumption in adjacent containers on the node. Whatever the reason, you may want to investigate what's going on within your workspace.

> We've now extended the JetBrains Gatweay [Backend Control Center](https://www.jetbrains.com/help/idea/work-inside-remote-project.html#control_center) with two additional bits of information about your workspace: `Workspace CPU` and `Workspace Memory`. The additional performance information shown in the control center is for the node that your workspace is running on, and not the workspace itself. This information is also the same that can be found from running the command `gp top` in your workspace.

Reviewer: we extended JetBrains backend not JetBrains Gatweay

Author: Thoughts on just removing "Gateway", adding "client" etc seems like it might be confusing. 

```suggestion
We've now extended the JetBrains [Backend Control Center](https://www.jetbrains.com/help/idea/work-inside-remote-project.html#control_center) with two additional bits of information about your workspace: `Workspace CPU` and `Workspace Memory`. The additional performance information shown in the control center is for the node that your workspace is running on, and not the workspace itself. This information is also the same that can be found from running the command `gp top` in your workspace.
```

The number of prompts you are now shown when opening VS Code Desktop is reduced.

You should see one browser redirect prompt (one for [VS Code Desktop](/docs/ides-and-editors/vscode) and one for [JetBrains Gateway](/docs/ides-and-editors/jetbrains-gateway)), marking "always allow" will suppress all future prompts. SSH fingerprints are also now automatically added to your `known_hosts` if you're using [SSH Gateway approach to opening VS Code Desktop](/docs/ides-and-editors/vscode#connecting-to-vs-code-desktop-ssh).

![Accepting the SSH fingerprint in VS Code](/images/editors/vscode-ssh-fingerprint-light-theme.png)
![Accepting the SSH fingerprint in VS Code](/images/editors/vscode-ssh-fingerprint-dark-theme.png)
_Accepting the SSH fingerprint in VS Code_

For more information, check out our dedicated documentation for [optimising VS Code Desktop](/docs/ides-and-editors/vscode#optimizing-vs-code-desktop).

> ## gp tasks list shows your current terminal

Reviewer: there is also `gp tasks stop` it is already deployed, but https://github.com/gitpod-io/website/pull/2639 should be merged

Author: Added in: b5a2fc34f3a4b1fa165375f160d92be3f5b84077, also approved related PR, indeed let's make sure that's merged before shipping this.

Author: Added note on main PR body: https://github.com/gitpod-io/website/pull/2578#issuecomment-1233164333

---
title: August Gitpod Release 2022
excerpt: TODO
date: 2022-09-01
image: 2022-09-01.jpg
alt: TODO
---

<script>
  import Contributors from "$lib/components/changelog/contributors.svelte";
</script>

TODO: Insert the featured changelog entry here.

<p><Contributors usernames="" /></p>

## JetBrains

## JetBrains now respects SDK versions defined in `.idea/misc.xml`

> Previously, when opening a Java Workspace with JetBrains, the Gitpod IDE would select the latest SDK version as the default. This was problematic when we recently updated the Gitpod `workspace-full` image, which many users are reliant upon, to include both Java 11 and 17. When having multiple SDK versions, JetBrains would default to using the latest SDK version, which in the case of the `workspace-full` image was 17. However, now Gitpod looks first at the `.idea/misc.xml`. If there is no SDK version set in `.idea/misc.xml`, the IDE will fall back to the `JAVA_HOME` environment variable.

Author: **Suggestion:** Would be good to add a note here about workspace-image versioning.

---
title: August Gitpod Release 2022
excerpt: The latest changes for Gitpod release in August 2022, including new commands for the `gp` CLI, updates to JetBrains to show multiple terminals in the IDE, and improvements to how VS Code Desktop connects to workspaces via SSH.
date: 2022-09-01
image: 2022-09-01.jpg
alt: TODO
---

> <script>

Reviewer: @loujaybee can we have highlights at the top and call out to latest releases as last time 🙏 We can use highlights this time as a shortcuts for different users to find relevant sections.

<img width="902" alt="Screenshot 2022-09-01 at 08 22 21" src="https://user-images.githubusercontent.com/3082655/187845742-00356348-4406-4fca-b09f-ca892078025c.png">

Author: Will have a go, a couple points to clarify: 

> can we have highlights at the top

To clarify, so we're on the same page: essentially a table of contents with tl;dr; for each item listed in this changelog with an anchor link to read more?

> call out to latest releases as last time 

I don't think we have any latest features mentioned directly in this changelog, do we? 

Would you like to carry through from last week e.g. the ports view in Desktop? Or are there additional "latest" features we wanted to highlight?

Author: VS Code added a table of contents (markdown plugin?) automatically, so I've left that in. Not sure how to configure header levels, ideally we'd just have H3's, H2's are quite verbose. Added a separate section for latest, linked ports view content from previous changelog, not sure if there's anything else we want to add here. 

![image](https://user-images.githubusercontent.com/5528307/187880032-456bedb6-cffd-47a0-a1a6-aeaa2429b9b9.png)

Let me know if you have thoughts on anything else to add (or updates), will resolve for now.  🙏

Reviewer: ah, ok let's drop latest 👍

  import Contributors from "$lib/components/changelog/contributors.svelte";
</script>

TODO: Insert the featured changelog entry here.

<p><Contributors usernames="" /></p>

> ## JetBrains

Author: @andreafalzetti, @felladrin can you review this section?

Reviewer 2: [Changes](https://gitpod.slack.com/archives/C01KGM9BH54/p1661965500447129?thread_ts=1661965022.653209&cid=C01KGM9BH54) needed for the Java version detection, but otherwise this section LGTM

We are continuing to make progress with the JetBrains and Gitpod integration #[7956](https://github.com/gitpod-io/gitpod/issues/7956) as part of our ongoing [partnership with JetBrains](https://www.gitpod.io/blog/gitpod-jetbrains). Whilst most of our efforts are mostly towards IntelliJ IDEA, a lot of the improvements we are making will still apply to our other [supported JetBrains IDEs](/docs/ides-and-editors).

### JetBrains multiple task Terminals

A key part of "Gitpodifying" a project is adding any necessary tasks, such as starting servers or database to your [.gitpod.yml](/docs/references/gitpod-yml). One advantage of declaring these processes in your `.gitpod.yml` configuration is that your commands are picked up by the IDE or editor and show on workspace start.

Previously, JetBrains with Gitpod only showed a single terminal, meaning that users needed to use `gp tasks` to interact with running terminals. Due to some improvements for JetBrains Remote Development that we are able to leverage, you can now see all of your terminals inside the IDE as you also can with VS Code.

### JetBrains now respects SDK versions defined in `.idea/misc.xml`

> Previously, when opening a Java Workspace with JetBrains, the Gitpod IDE would select the latest SDK version as the default. This was problematic when we recently updated the Gitpod `workspace-full` image, which many users are reliant upon, to include both Java 11 and 17. When having multiple SDK versions, JetBrains would default to using the latest SDK version, which in the case of the `workspace-full` image was 17. However, now Gitpod looks first at the `.idea/misc.xml`. If there is no SDK version set in `.idea/misc.xml`, the IDE will fall back to the `JAVA_HOME` environment variable.

Reviewer: - I would replace `.idea/misc.xml` with something like if a project does not have yet configured JDK.
- We look not only at JAVA_HOME but generally at user environments to detect properly default JDK, i.e. PATH variable and so on.

> Previously, when opening a Java Workspace with JetBrains, the Gitpod IDE would select the latest SDK version as the default. This was problematic when we recently updated the Gitpod `workspace-full` image, which many users are reliant upon, to include both Java 11 and 17. When having multiple SDK versions, JetBrains would default to using the latest SDK version, which in the case of the `workspace-full` image was 17. However, now Gitpod looks first at the `.idea/misc.xml`. If there is no SDK version set in `.idea/misc.xml`, the IDE will fall back to the `JAVA_HOME` environment variable.

Reviewer: 👍 for mentioning that we ship both Java versions is workspace-full and java 11 as default

> Previously, when opening a Java Workspace with JetBrains, the Gitpod IDE would select the latest SDK version as the default. This was problematic when we recently updated the Gitpod `workspace-full` image, which many users are reliant upon, to include both Java 11 and 17. When having multiple SDK versions, JetBrains would default to using the latest SDK version, which in the case of the `workspace-full` image was 17. However, now Gitpod looks first at the `.idea/misc.xml`. If there is no SDK version set in `.idea/misc.xml`, the IDE will fall back to the `JAVA_HOME` environment variable.

Reviewer 3: **Suggestion:** replace "SDK" by "Java SDK (JDK)" because this change doesn't affect any other language SDK other than Java.

Reviewer: yes, it is only specifically for IntelliJ IDEA

Author: Ah, yes that's a great point, thank you! Updated in: d846465203ef72271153ce2cf4b43fc2d14d4227

![](/images/jetbrains-gateway/java-sdk-configuration.png)
_JetBrains IDEs now picking up the correct SDK version by default_

### View Resource Consumption in JetBrains Backend Control Center

When using a Gitpod workspace you might experience performance issues caused by an application using more resources than expected, or due to resource consumption in adjacent containers on the node. Whatever the reason, you may want to investigate what's going on within your workspace.

We've now extended the JetBrains [Backend Control Center](https://www.jetbrains.com/help/idea/work-inside-remote-project.html#control_center) with two additional bits of information about your workspace: `Workspace CPU` and `Workspace Memory`. The additional performance information shown in the control center is for the node that your workspace is running on, and not the workspace itself. This information is also the same that can be found from running the command `gp top` in your workspace.

![](/images/jetbrains-gateway/jetbrains-performance-center.png)
_Viewing the workspace and node performance in the Backend Control Center_

> ## VS Code Desktop

Author: @jeanp413 / @filiptronicek please can you review this section?

Reviewer 4: LGTM

> We have been continuing to invest in our desktop editor experiences, including VS Code #[7184](https://github.com/gitpod-io/gitpod/issues/7184), with an emphasis on removing any barriers or friction in getting the VS Code editor open with a Gitpod workspace, establishing a stable connection via SSH, and improving the experience of restarting a timed out workspace.

Reviewer 5: > restarting a timed out workspace

I think I missed this one, @loujaybee can you link anything about this? 

The rest summarizes well what we are trying to achieve here: stability

Author: There were some issues due to the host name being dynamic (e.g. when using the access token approach for SSH) where you lost some VS Code workspace layouts, due to them being tied to the host name. 

But also when your workspace times out the access token would rotate and you'd have to re-authenticate. I think in most cases that was forcing users to go back to the dashboard and re-open the workspace rather than simply re-connecting. We now have a [section in the docs on VS Code Desktop re-connecting](https://www.gitpod.io/docs/ides-and-editors/vscode#reconnect-to-vs-code-desktop), not sure if it's worth 
documenting too much the old behaviours, but more opportunities to improve here e.g. 

- https://github.com/gitpod-io/gitpod/issues/11701
- https://github.com/gitpod-io/gitpod/issues/11263
- https://github.com/gitpod-io/gitpod/issues/7685

![Accepting the SSH fingerprint in VS Code](/images/editors/vscode-ssh-fingerprint-dark-theme.png)
_Using the SSH Gateway approach for VS Code Desktop means no more SSH fingerprint prompts_

For more information, check out our dedicated documentation for [optimising VS Code Desktop](/docs/ides-and-editors/vscode#optimizing-vs-code-desktop).

### All VS Code Desktop users on SaaS are now using SSH Gateway! 🎉

We have now completed a migration of all VS Code Desktop users on SaaS to our new SSH Gateway approach. This change brings improved stability, smoother restarting of workspaces, and more.

To get setup, we recommend you [upload an SSH key to Gitpod](https://www.gitpod.io/docs/configure/ssh#upload-an-ssh-key-to-gitpod). For full details on why we made this change to our SSH approach for VS Code Desktop, please check out our blog post [VS Code Desktop and SSH explained](https://www.gitpod.io/blog/vscode-desktop-ssh-updates).

If you cannot use the new approach for any reason, please reach out via the GitHub issue #[7452](https://github.com/gitpod-io/gitpod/issues/7452).

![](/images/docs/new-ssh-key-light-theme.png)

> ## The Gitpod CLI (`gp`)

Author: @andreafalzetti - can you review this section? 🙏

Reviewer 2: Magnificent write-up, as always!

Let's say that you're documenting your Gitpod setup, rather than detailing actions directly how to take some action in each of the [Gitpod supported IDEs & editors](/docs/ides-and-editors), you can share `gp` commands knowing that the experience will be consistent across all workspace editing experiences.

### `gp tasks list` shows your current tasks (as defined in the `.gitpod.yml`)

The `gp tasks list` command allows you to list any running commands that are defined in your `.gitpod.yml`. If you want, you can even attach to a given task using `gp tasks attach` which brings that current task into the foreground of your terminal. However, since there are situations where it might not be as easy to figure out which terminal you're currently viewing, that's why we now highlight your current task to make life a little easier for you

![](/images/editors/gp-task-list-highlight.png)

### `gp tasks stop` stops running tasks that (as defined in the `.gitpod.yml`)

> You can now run `gp tasks stop` to stop running tasks that have been defined in your `.gitpod.yml`. You can either stop a single task with `gp tasks stop <id`, or terminate all tasks by running `gp tasks stop --all`.

Reviewer 2: ```suggestion
You can now run `gp tasks stop` to stop running tasks that have been defined in your `.gitpod.yml`. You can either stop a single task with `gp tasks stop <id>`, or terminate all tasks by running `gp tasks stop --all`.
```

See the `gp` [Command Line Interface](/docs/command-line-interface#stop-1) documentation for more.

<!-- TODO: Add about multiple tasks in JetBrains view -->

## SSH

The SSH protocol is the primary method of connection to Gitpod workspaces for JetBrains IDEs, VS Code Desktop and users who directly SSH into Gitpod for editing, or working with their code. We've been working to ensure that Gitpod has a wide range of support for different SSH clients and SSH key versions.

> ### Improved OpenSSH compatibility and support for sha256/sha512

Author: @iQQBot - Can you review this section? 🙏

---
title: August Gitpod Release 2022
excerpt: The latest changes for Gitpod release in August 2022, including new commands for the `gp` CLI, updates to JetBrains to show multiple terminals in the IDE, and improvements to how VS Code Desktop connects to workspaces via SSH.
date: 2022-09-01

> image: 2022-09-01.jpg

Reviewer 6: What is the location of this file? I think, this is causing problem: [netlify logs](https://app.netlify.com/sites/gitpod-kumquat/deploys/6310bfee6a46420009fec569#L2401)

Author: Pushing a temp commit with a valid path ee738189718874a2ef280d9f3b57b73bc4f8ff69 builds for me in a workspace, should start passing 🙏 🤞

> image: 2022-08-26.jpg

Author: **Note:** This needs updating with a new header image for this changelog.

Author: Added in: abdfb46d965f52147388c592f11008e285a4c71a

- [#12096](https://github.com/gitpod-io/gitpod/pull/12096) - Update spending limit modal on workspace start <Contributors usernames="geropl,gtsiolis,jankeromnes,jldec,laushinka" />
- [#12234](https://github.com/gitpod-io/gitpod/pull/12234) - Truncate project environment variable name <Contributors usernames="geropl,gtsiolis" />
- [#12163](https://github.com/gitpod-io/gitpod/pull/12163) - JetBrains IDEs will use the default Java SDK version when there's no explicit configuration for it yet. <Contributors usernames="akosyakov,andreafalzetti,felladrin" />
- [#12028](https://github.com/gitpod-io/gitpod/pull/12028) - Fixed a rare case in which prebuilds terminated successfully but were shown as failed <Contributors usernames="jenting,sagor999,utam0k" />
- [#12173](https://github.com/gitpod-io/gitpod/pull/12173) - [installer]: remove all podsecuritypolicies <Contributors usernames="MrSimonEmms,aledbf,corneliusludmann,csweichel,jenting" />
- [#12280](https://github.com/gitpod-io/gitpod/pull/12280) - Update docker compose from v2.8.0 to v2.10.0 <Contributors usernames="aledbf,sagor999" />

> - [#12313](https://github.com/gitpod-io/gitpod/pull/12313) - Revert "[installer]: remove all podsecuritypolicies" #12313 <Contributors usernames="MrSimonEmms,adrienthebo,sagor999" />

Author: Do we want to include reverts in the changelog? 🤔

Reviewer 7: eeeh probably not. I think you can remove the two above (#12173 and #12313), and only keep #12340 which seems to be how we solved it in the end. Context: The events seem to have been: remove, revert, add deprecation warning. Based on that I'd just keep the last one.

Author: @atduarte has got it, looks like: 71e8eadb57be815e9ab0ff060ce1f52ffa7ee7a6 😄

Reviewer 8: It was also about the installer so :P

---

> title: August Gitpod Release 2022

Author: August, or September 🤔 💭

Reviewer: I think it should be August

### JetBrains now supports multiple task Terminals

A key part of "Gitpodifying" a project is adding any necessary tasks, such as starting servers or database to your [.gitpod.yml](/docs/references/gitpod-yml). One advantage of declaring these processes in your `.gitpod.yml` configuration is that your commands are picked up by the IDE or editor and show on workspace start.

Previously, JetBrains with Gitpod only showed a single terminal, meaning that users needed to use `gp tasks` to interact with running terminals. Due to some improvements for JetBrains Remote Development that we are able to leverage, you can now see all of your terminals inside the IDE as you also can with VS Code.

### JetBrains now respects configured Java SDK (JDK) versions

Previously, when opening a Java Workspace with JetBrains, the Gitpod IDE would select the latest Java SDK (JDK) version as the default.

> This was problematic when we recently updated the Gitpod `workspace-full` image, to include both Java 11 and 17, which many users rely upon. When having multiple Java SDK (JDK) versions, JetBrains would default to using the latest Java SDK (JDK) version, which in the case of the `workspace-full` image was 17, and not 11, causing a minor regression. Now, Gitpod respects configured Java SDK (JDK) versions in locations such as the `PATH` and `JAVA_HOME`.

Reviewer 3: Thanks for the update, @loujaybee!

I think on this second paragraph we can refer to "JDK" directly, since it's been defined is on the previous one.
```suggestion
This was problematic when we recently updated the Gitpod `workspace-full` image, to include both Java 11 and 17, which many users rely upon. When having multiple JDK versions, JetBrains would default to using the latest JDK version, which in the case of the `workspace-full` image was 17, and not 11, causing a minor regression. Now, Gitpod respects configured JDK versions in locations such as the `PATH` and `JAVA_HOME`.
```

> This was problematic when we recently updated the Gitpod `workspace-full` image, to include both Java 11 and 17, which many users rely upon. When having multiple Java SDK (JDK) versions, JetBrains would default to using the latest Java SDK (JDK) version, which in the case of the `workspace-full` image was 17, and not 11, causing a minor regression. Now, Gitpod respects configured Java SDK (JDK) versions in locations such as `.idea/misc.xml` and `JAVA_HOME`.

Author: ```suggestion
This was problematic when we recently updated the Gitpod `workspace-full` image, to include both Java 11 and 17, which many users rely upon. When having multiple Java SDK (JDK) versions, JetBrains would default to using the latest Java SDK (JDK) version, which in the case of the `workspace-full` image was 17, and not 11, causing a minor regression. Now, Gitpod respects configured Java SDK (JDK) versions in locations such as the `PATH` and `JAVA_HOME`.
```

See the `gp` [Command Line Interface](/docs/command-line-interface#stop-1) documentation for more.

<!-- TODO: Add about multiple tasks in JetBrains view -->

## Improvements for SSH connections to Gitpod

The SSH protocol is the primary method of connection to Gitpod workspaces for JetBrains IDEs, VS Code Desktop and users who directly SSH into Gitpod for editing, or working with their code. We've been working to ensure that Gitpod has a wide range of support for different SSH clients and SSH key versions.

### Improved OpenSSH compatibility and support for sha256/sha512

Previously, Gitpod did not support OpenSSH clients >v8.8, nor did Gitpod support RSA sha256/sha512. For users using VS Code Desktop with the SSH Gateway approach, this could mean that in some cases users could not leverage SSH access to Gitpod due to incompatible keys or OpenSSH client's, however this issue should now be resolved.

## Other fixes and improvements

- [#12468](https://github.com/gitpod-io/gitpod/pull/12468) - [local-preview] Check and exit if M1 Mac <Contributors usernames="Pothulapati,corneliusludmann" />

> - [#12116](https://github.com/gitpod-io/gitpod/pull/12116) - Gitpod CLI: New command `gp tasks stop <Contributors usernames="akosyakov,andreafalzetti,felladrin,filiptronicek,mustard-mh" />

Reviewer 8: ```suggestion
- [#12116](https://github.com/gitpod-io/gitpod/pull/12116) - Gitpod CLI: New command `gp tasks stop` <Contributors usernames="akosyakov,andreafalzetti,felladrin,filiptronicek,mustard-mh" />
```

> - [#12422](https://github.com/gitpod-io/gitpod/pull/12422) - experimental: enable the feature flag PVC, the files/folders under .git/ folder user:group permission is incorrect <Contributors usernames="aledbf,andreafalzetti,jenting,mustard-mh,sagor999" />

Reviewer 8: ```suggestion
```

> - [#11255](https://github.com/gitpod-io/gitpod/pull/11255) - Limit the rate at which network connections can be made by a workspace <Contributors usernames="Furisto,MrSimonEmms,aledbf,geropl,jenting,kylos101,sagor999,utam0k" />

Reviewer 8: ```suggestion
- [#11255](https://github.com/gitpod-io/gitpod/pull/11255) - Prevent abuse by limiting rate at which network connections can be made by a workspace <Contributors usernames="Furisto,MrSimonEmms,aledbf,geropl,jenting,kylos101,sagor999,utam0k" />
```

- [#12258](https://github.com/gitpod-io/gitpod/pull/12258) - Added phone number verification in SaaS, to help mitigate abuse. <Contributors usernames="AlexTugarev,corneliusludmann,gtsiolis,jldec,kylos101,svenefftinge" />

> - [#12398](https://github.com/gitpod-io/gitpod/pull/12398) - [kots] load dockerConfigJson reigstry names into `privateBaseImageAllowList <Contributors usernames="MrSimonEmms,Pothulapati,lucasvaltl,nandajavarma" />

Reviewer 8: Should this be removed and only be part of the self-hosted changelog? I don't remember what's the guideline right now.

Author: CC: @lucasvaltl

Reviewer 8: Removed 🙏

> - [#12417](https://github.com/gitpod-io/gitpod/pull/12417) - Upgrade runc to 1.1.4 <Contributors usernames="aledbf,utam0k" />

Reviewer 8: ```suggestion
```

> - [#12367](https://github.com/gitpod-io/gitpod/pull/12367) - More reliably connect to buildkit during image builds <Contributors usernames="jenting,utam0k" />

Reviewer 8: ```suggestion
- [#12367](https://github.com/gitpod-io/gitpod/pull/12367) - Improve reliability of image build starts<Contributors usernames="jenting,utam0k" />
```

- [#12340](https://github.com/gitpod-io/gitpod/pull/12340) - [installer]: show deprecation warning if using podsecuritypolicies <Contributors usernames="MrSimonEmms,corneliusludmann" />
- [#12336](https://github.com/gitpod-io/gitpod/pull/12336) - [installer]: conditionally remove pod security policies <Contributors usernames="MrSimonEmms,jenting,nandajavarma" />

> - [#12315](https://github.com/gitpod-io/gitpod/pull/12315) - Prevent divide by zero error in workspace info service <Contributors usernames="Furisto,jenting,kylos101,utam0k" />

Reviewer 8: Not yet deployed 😬 

How can we improve the automation to take into consideration the "deployed" label? Also how can we avoid forgetting this entry in the next changelog?

Author: [This is the current script](https://github.com/gitpod-io/website/blob/main/scripts/generate-changelog.js), would need to be extended or updated, possibly. I think we should [open an issue](https://github.com/gitpod-io/website/issues/new), seems we already have a tag for changelog issues (`section: changelog`). Could be something @Cahllagerfeld and website team could help with, but until then we'll have to stick with manual verification.

Reviewer 8: Let's remove it for now

```suggestion
```

- [#12288](https://github.com/gitpod-io/gitpod/pull/12288) - [ssh-gateway] support rsa sha256/sha512 algorithm <Contributors usernames="MrSimonEmms,akosyakov,iQQBot,jeanp413,sagor999" />
- [#12096](https://github.com/gitpod-io/gitpod/pull/12096) - Update spending limit modal on workspace start <Contributors usernames="geropl,gtsiolis,jankeromnes,jldec,laushinka" />
- [#12234](https://github.com/gitpod-io/gitpod/pull/12234) - Truncate project environment variable name <Contributors usernames="geropl,gtsiolis" />
- [#12163](https://github.com/gitpod-io/gitpod/pull/12163) - JetBrains IDEs will use the default Java SDK version when there's no explicit configuration for it yet. <Contributors usernames="akosyakov,andreafalzetti,felladrin" />

> - [#12028](https://github.com/gitpod-io/gitpod/pull/12028) - Fixed a rare case in which prebuild was terminating successfully but failing <Contributors usernames="jenting,sagor999,utam0k" />

Reviewer 8: ```suggestion
- [#12028](https://github.com/gitpod-io/gitpod/pull/12028) - Fixed a rare case in which prebuilds terminated successfully but were shown as failed <Contributors usernames="jenting,sagor999,utam0k" />
```

- [#12173](https://github.com/gitpod-io/gitpod/pull/12173) - [installer]: remove all podsecuritypolicies <Contributors usernames="MrSimonEmms,aledbf,corneliusludmann,csweichel,jenting" />

> - [#12280](https://github.com/gitpod-io/gitpod/pull/12280) - Update docker compose to v2.10.0 <Contributors usernames="aledbf,sagor999" />

Reviewer 8: Maybe this should be in higher evidence. More to the top of the list.

```suggestion
- [#12280](https://github.com/gitpod-io/gitpod/pull/12280) - Update docker compose from v2.8.0 to v2.10.0 <Contributors usernames="aledbf,sagor999" />
```

---
title: TODO
excerpt: TODO
date: 2022-09-01
image: 2022-09-01.jpg
alt: TODO
---

<script>
  import Contributors from "$lib/components/changelog/contributors.svelte";
</script>

TODO: Insert the featured changelog entry here.

<p><Contributors usernames="" /></p>

## VS Code Desktop improved opening flow

<!-- TODO: Discuss popup optimisation -->

## JetBrains performance metrics

<!-- TODO: Discuss performance monitoring addition -->

## JetBrains features from Latest to Stable

<!-- TODO: Add about multiple tasks in JetBrains view -->

### Fixes and improvements

- [#12468](https://github.com/gitpod-io/gitpod/pull/12468) - [local-preview] Check and exit if M1 Mac <Contributors usernames="Pothulapati,corneliusludmann" />

> - [#12116](https://github.com/gitpod-io/gitpod/pull/12116) - Gitpod CLI: New command `gp tasks stop <Contributors usernames="akosyakov,andreafalzetti,felladrin,filiptronicek,mustard-mh" />

Reviewer: @andreafalzetti what is preventing to merge https://github.com/gitpod-io/website/pull/2639?

@loujaybee this one with gp top [[1](https://github.com/gitpod-io/gitpod/pull/12338#issuecomment-1230304284)][[2](https://github.com/gitpod-io/website/pull/2653)] can be `GP Cli` section

Author: It seems `gp top --json` is not yet deployed and documentation still pending: 

- https://github.com/gitpod-io/website/pull/2653/files

Reviewer 2: I was just waiting for at least one approval, it has now been merged. Thanks @loujaybee :)

Author: Will resolve, we can put the `--json` fix into the next changelog 🙏

- [#12398](https://github.com/gitpod-io/gitpod/pull/12398) - [kots] load dockerConfigJson reigstry names into `privateBaseImageAllowList <Contributors usernames="MrSimonEmms,Pothulapati,lucasvaltl,nandajavarma" />
- [#12417](https://github.com/gitpod-io/gitpod/pull/12417) - Upgrade runc to 1.1.4 <Contributors usernames="aledbf,utam0k" />
- [#12367](https://github.com/gitpod-io/gitpod/pull/12367) - More reliably connect to buildkit during image builds <Contributors usernames="jenting,utam0k" />
- [#12340](https://github.com/gitpod-io/gitpod/pull/12340) - [installer]: show deprecation warning if using podsecuritypolicies <Contributors usernames="MrSimonEmms,corneliusludmann" />
- [#12336](https://github.com/gitpod-io/gitpod/pull/12336) - [installer]: conditionally remove pod security policies <Contributors usernames="MrSimonEmms,jenting,nandajavarma" />
- [#12315](https://github.com/gitpod-io/gitpod/pull/12315) - Prevent divide by zero error in workspace info service <Contributors usernames="Furisto,jenting,kylos101,utam0k" />

> - [#12288](https://github.com/gitpod-io/gitpod/pull/12288) - [ssh-gateway] support rsa sha256/sha512 algorithm <Contributors usernames="MrSimonEmms,akosyakov,iQQBot,jeanp413,sagor999" />

Reviewer: @loujaybee this one should be `SSH` section, mentioning that we support know lates Open SSH clients

cc @iQQBot it is version 9.x?

Author: Added in: c12d21d6030d6336156aebfc6b6e7956a68e78a1

- [#12096](https://github.com/gitpod-io/gitpod/pull/12096) - Update spending limit modal on workspace start <Contributors usernames="geropl,gtsiolis,jankeromnes,jldec,laushinka" />
- [#12234](https://github.com/gitpod-io/gitpod/pull/12234) - Truncate project environment variable name <Contributors usernames="geropl,gtsiolis" />

> - [#12163](https://github.com/gitpod-io/gitpod/pull/12163) - JetBrains IDEs will use the default Java SDK version when there's no explicit configuration for it yet. <Contributors usernames="akosyakov,andreafalzetti,felladrin" />

Reviewer: It can go under `JetBrains` section explain what we correctly detect and configure default JDK now

Author: Added in: 8c50c1f2375363f3f79e8509670ac83343e54916

But I have some questions ([asking internally](https://gitpod.slack.com/archives/C01KGM9BH54/p1661965022653209))

- [#12028](https://github.com/gitpod-io/gitpod/pull/12028) - Fixed a rare case in which prebuild was terminating successfully but failing <Contributors usernames="jenting,sagor999,utam0k" />
- [#12173](https://github.com/gitpod-io/gitpod/pull/12173) - [installer]: remove all podsecuritypolicies <Contributors usernames="MrSimonEmms,aledbf,corneliusludmann,csweichel,jenting" />
- [#12280](https://github.com/gitpod-io/gitpod/pull/12280) - Update docker compose to v2.10.0 <Contributors usernames="aledbf,sagor999" />
- [#12313](https://github.com/gitpod-io/gitpod/pull/12313) - Revert "[installer]: remove all podsecuritypolicies" #12313 <Contributors usernames="MrSimonEmms,adrienthebo,sagor999" />
- [#12227](https://github.com/gitpod-io/gitpod/pull/12227) - [kots]: give installer same node affinity as ws-daemon <Contributors usernames="MrSimonEmms,nandajavarma" />
- [#12267](https://github.com/gitpod-io/gitpod/pull/12267) - [installer]: add golden files for aws, azure and gcp <Contributors usernames="MrSimonEmms,adrienthebo" />

> - [#12159](https://github.com/gitpod-io/gitpod/pull/12159) - JetBrains: Display Workspace CPU/Memory usage in Backend Control Center.JetBrains: Workspaces created from repositories with tasks defined on .gitpod.yml will start with one terminal opened for each task. <Contributors usernames="akosyakov,andreafalzetti,felladrin" />

Reviewer: It should go under `JetBrains` section too. I would explain that users should have a look at it, and then use the last tab to change Xmx and if it works then persist settings into `jetbrains.[product].vmoptions`

Reviewer: Also that we know show all terminals.

Author: Should be covered in: 

- https://github.com/gitpod-io/website/pull/2672/files

- [#12174](https://github.com/gitpod-io/gitpod/pull/12174) - [kots] Allow users to upload a `.docker/config.json` file <Contributors usernames="MrSimonEmms,Pothulapati,lucasvaltl" />
- [#12218](https://github.com/gitpod-io/gitpod/pull/12218) - [kots] show `registry_s3` options only when `incluster` is enabled <Contributors usernames="MrSimonEmms,Pothulapati" />
- [#12211](https://github.com/gitpod-io/gitpod/pull/12211) - [kots]: remove help text documenting the limitation on custom labels <Contributors usernames="MrSimonEmms,adrienthebo" />
- [#12139](https://github.com/gitpod-io/gitpod/pull/12139) - Mark workspace's whose image builds failed as `stopped <Contributors usernames="geropl,laushinka" />
- [#11971](https://github.com/gitpod-io/gitpod/pull/11971) - [kots]: allow for backup of resources in KOTS <Contributors usernames="MrSimonEmms,corneliusludmann" />
- [#12043](https://github.com/gitpod-io/gitpod/pull/12043) - [kots]: add useful links to the KOTS dashboard <Contributors usernames="MrSimonEmms,corneliusludmann,lucasvaltl" />

> - [#12082](https://github.com/gitpod-io/gitpod/pull/12082) - Enable allowing redirect to Desktop IDEs for all workspaces. <Contributors usernames="akosyakov,felladrin,loujaybee" />

Reviewer: It is VS Code Desktop improved opening flow

Author: Added!

> - [#12053](https://github.com/gitpod-io/gitpod/pull/12053) - JetBrains: Display Workspace CPU/Memory usage in Backend Control Center <Contributors usernames="akosyakov,andreafalzetti,felladrin" />

Reviewer: it is `JetBrains performance metrics`

Author: Added!

> - [#12052](https://github.com/gitpod-io/gitpod/pull/12052) - Gitpod CLI: Highlight current task in gp tasks list <Contributors usernames="akosyakov,andreafalzetti,felladrin,filiptronicek,jeanp413" />

Reviewer: Also should go under `GP cli`

Author: Added!

- [#11983](https://github.com/gitpod-io/gitpod/pull/11983) - N/A <Contributors usernames="jenting,utam0k" />
- [#11954](https://github.com/gitpod-io/gitpod/pull/11954) - [installer]: remove the custom labels from the selector labels <Contributors usernames="MrSimonEmms,aledbf,andrew-farries,cooperbenson-qz,corneliusludmann,felladrin" />
- [#11993](https://github.com/gitpod-io/gitpod/pull/11993) - Fixes entering of colons in host on Git Integration page. <Contributors usernames="AlexTugarev,andrew-farries" />
- [#12037](https://github.com/gitpod-io/gitpod/pull/12037) - [installer]: escape inline yaml keys for labels/annotations in Helm <Contributors usernames="MrSimonEmms,adrienthebo" />
- [#12031](https://github.com/gitpod-io/gitpod/pull/12031) - [installer]: use typemeta declaration in the helm customisations <Contributors usernames="MrSimonEmms,nandajavarma" />

> - [#11081](https://github.com/gitpod-io/gitpod/pull/11081) - All open ports from the workspace are automatically forwarded when using Latest JetBrains IDEs. <Contributors usernames="akosyakov,aledbf,andreafalzetti,felladrin,meysholdt,mustard-mh" />

Reviewer: It should be removed, it is only in latest. cc @felladrin please mark next time

Author: ```suggestion
```

- [#11980](https://github.com/gitpod-io/gitpod/pull/11980) - N/A <Contributors usernames="andrew-farries,jenting" />
- [#11965](https://github.com/gitpod-io/gitpod/pull/11965) - [kots]: fix typo which meant wasn't using the appSlug in config collect <Contributors usernames="MrSimonEmms,Pothulapati" />
- [#11943](https://github.com/gitpod-io/gitpod/pull/11943) - [kots]: collect the user config in support bundles <Contributors usernames="MrSimonEmms,Pothulapati" />
- [#11898](https://github.com/gitpod-io/gitpod/pull/11898) - `gpctl workspaces list` shows all workspace types <Contributors usernames="jenting,utam0k" />
- [#11576](https://github.com/gitpod-io/gitpod/pull/11576) - Add Spending Limit Reached modal on workspace creation. <Contributors usernames="AlexTugarev,geropl" />
- [#10199](https://github.com/gitpod-io/gitpod/pull/10199) - Migrate AlertBox component instances to Alert component <Contributors usernames="KevSlashNull,andrew-farries,gtsiolis,meysholdt" />
- [#11878](https://github.com/gitpod-io/gitpod/pull/11878) - [kots]: add docker.io to registry allowlist <Contributors usernames="MrSimonEmms,mrzarquon,nandajavarma" />

> - [#11900](https://github.com/gitpod-io/gitpod/pull/11900) - [code] update stable VSCode Browser to 1.70.0 <Contributors usernames="corneliusludmann,iQQBot,mustard-mh" />

Reviewer: I think we can remove it.

Author: ```suggestion
```

> - [#11877](https://github.com/gitpod-io/gitpod/pull/11877) - Fix in-product changelog relative URLs not working problem <Contributors usernames="MrSimonEmms,akosyakov,mustard-mh" />

Reviewer: We can also remove it.

Author: ```suggestion
```
